source/async_stack.cpp

```diff
@@ -58,34 +63,49 @@ struct AsyncStackRootHolderList {
     asyncStackRootHolders_.reserve(50);
   }
 
-void add(void* holder) {
-  std::lock_guard<std::mutex> lock(mutex_);
-  asyncStackRootHolders_.push_back(holder);
-}
+  void add(void* holder) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    asyncStackRootHolders_.push_back(holder);
+  }
 
-void remove(void* holder) {
-  std::lock_guard<std::mutex> lock(mutex_);
-  auto it = std::find(asyncStackRootHolders_.begin(), asyncStackRootHolders_.end(), holder);
-  if (it != asyncStackRootHolders_.end()) {
-    std::swap(*it, asyncStackRootHolders_.back());
-    asyncStackRootHolders_.pop_back();
+  void remove(void* holder) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    auto it = std::find(
+        asyncStackRootHolders_.begin(), asyncStackRootHolders_.end(), holder);
+    if (it != asyncStackRootHolders_.end()) {
+      std::swap(*it, asyncStackRootHolders_.back());
+      asyncStackRootHolders_.pop_back();
+    }
   }
-}
 
-std::vector<void*> getAsyncStackRoots() {
-  if (!mutex_.try_lock()) {
-    // assume we crashed holding the lock and give up
-    return {};
+  std::vector<void*> getAsyncStackRoots() {
+    if (!mutex_.try_lock()) {
+      // assume we crashed holding the lock and give up
+      return {};
+    }
+    std::lock_guard<std::mutex> lock(mutex_, std::adopt_lock);
+    return asyncStackRootHolders_;
   }
-  std::lock_guard<std::mutex> lock(mutex_, std::adopt_lock);
-  return asyncStackRootHolders_;
-}
 };
 
 extern "C" {
-  auto* kUnifexAsyncStackRootHolderList = new AsyncStackRootHolderList();
+auto* kUnifexAsyncStackRootHolderList = new AsyncStackRootHolderList();
+}
+
+namespace utils {
+static std::uint64_t get_os_thread_id() {
+#  if defined(__APPLE__)
+  std::uint64_t tid;
+  pthread_threadid_np(nullptr, &tid);
+  return tid;
+#  elif defined(_WIN32)
+  return std::uint64_t(GetCurrentThreadId());
+#  else
+  return std::uint64_t(gettid());
+#  endif
 }
-#endif // UNIFEX_ASYNC_STACK_ROOT_USE_PTHREAD == 0
+}  // namespace utils
```

Comment:
Since `get_os_thread_id` is static and inside the `unifex` namespace, I don't think you don't need the `utils` namespace.